File: docs/snippets/all/archetypes/scalars_row_updates.py
"""
Update a scalar over time.

See also the `scalar_column_updates` example, which achieves the same
thing in a single operation.
"""

from __future__ import annotations

import math

import dalaran as dl

dl.init("dalaran_example_scalar_row_updates", spawn=True)

for step in range(64):
    dl.set_time("step", sequence=step)
    dl.log("scalars", dl.Scalars(math.sin(step / 10.0)))

File: docs/snippets/all/archetypes/scalars_row_updates.rs
//! Update a scalar over time.
//!
//! See also the `scalar_column_updates` example, which achieves the same thing in a single operation.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec =
        dalaran::RecordingStreamBuilder::new("dalaran_example_scalar_row_updates")
            .spawn()?;

    for step in 0..64 {
        rec.set_time_sequence("step", step);
        rec.log(
            "scalars",
            &dalaran::Scalars::single((step as f64 / 10.0).sin()),
        )?;
    }

    Ok(())
}
